### PR TITLE
New closure type ClosurePtr<P, R> with hooks

### PR DESCRIPTION
**Pull request by namse**
This is the PR that changes attach_event's closure parameter to ClosurePtr<P, R> which always return true on eq().
ClosurePtr keeps the pipe to closure and when user change the endpoint-closure, it also changes the target to invoke automatically. It's kind of linked-list.

The purpose of ClosurePtr is to prevent re-rendering when event handler changed.

I added the hooks sample which takes the big benefit of ClosurePtr's always-return-true-on-eq, because hooks compare props when it re-render the component.

Because of changes and dumb rustc, sometime we need to put type annotation for closure. It's easy to put on vscode using `ctrl + .` so I don't worry about it.

### CODE REVIEW

File: namui/sample/hooks/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "rust-analyzer.linkedProjects": ["./Cargo.toml", "./Cargo.toml"]
+}
```
Comment (bigfoodK) [inline namui/sample/hooks/.vscode/settings.json:3]: What does it means?

Comment (namse) [reply namui/sample/hooks/.vscode/settings.json:3]: I... have no idea

File: namui/sample/hooks/rust-toolchain.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly"
```
Comment (bigfoodK) [inline namui/sample/hooks/rust-toolchain.toml:2]: Hooks are available only in nightly?

Comment (namse) [reply namui/sample/hooks/rust-toolchain.toml:2]: No, it's not.

File: namui/sample/hooks/.vscode/settings.json (hunk 1 above)
Comment (namse) [reply namui/sample/hooks/.vscode/settings.json]: I removed this.

File: namui/sample/hooks/rust-toolchain.toml (hunk 2 above)
Comment (namse) [reply namui/sample/hooks/rust-toolchain.toml]: Removed

**Review by bigfoodK — APPROVED** [review]

**Pull request merged**